back/engines/commercial/report_builder/spec/policies/report_builder/report_policy_spec.rb

```diff
@@ -21,6 +21,33 @@
     it { expect(scope.resolve.count).to eq(3) }
   end
 
+  context 'when user has moderator rights' do
+    let_it_be(:user) { build(:project_moderator) }
+
+    context 'when user owns the report' do
+      let_it_be(:all_reports) { create_list(:report, 3) }
+      let_it_be(:report) do
+        all_reports.first.tap { |r| r.update!(owner: user) }
+      end
+
+      it { is_expected.to permit(:show) }
+      it { is_expected.to permit(:layout) }
+      it { is_expected.to permit(:create) }
+      it { is_expected.to permit(:destroy) }
+      it { is_expected.to permit(:update) }
+      it { expect(scope.resolve.count).to eq(1) }
+    end
+
+    context 'when user does not own the report' do
```

Comment:
But when the report has a phase context, the moderator should always be able to view the report, and if the report does not contain data the moderator does not have access to they should also be permitted to do all the other actions. Should we also test explicitly for those cases? Or are we already testing those cases somewhere else?

Reply:
Good point, it's indeed not tested. Let me do it. 

BTW the relation with the action descriptor will be handled in this WIP PR https://github.com/CitizenLabDotCo/citizenlab/pull/6884/files

Reply:
@luucvanderzee Done.